src/components/text-input/TextInput.tsx

```diff
@@ -75,6 +75,14 @@ export const TextField = forwardRef<HTMLInputElement, PropsWithLabels<Props>>((p
   const defaultId = `${useId()}textfield`;
   const controlId = id ?? defaultId;
   const descriptionId = `${controlId}-description`;
+  const validationMessageId = `${controlId}-validation-message`;
+
+  const describedBy = [
+    Boolean(messageValid || messageInvalid) && validationMessageId,
+    Boolean(helpText) && descriptionId,
+  ]
+    .filter((id) => id)
+    .join(' ');
```

Comment:
Should we extract it to a hook reused in each components? We could also put that piece in the hook: `describedBy !== '' ? describedBy : undefined`

Reply:
I'll see if I can come up with a hook API that makes sense. This is why I wanted to use a FormField context, so all the logic could be in the form field and the controls could just do something like:

```typescript
const { controlId, describedBy, validation } = useContext(FormFeildContext);
```